internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -317,6 +318,9 @@ func (r *PersistentVolumeClaimReconciler) storageClassEventHandler() handler.Eve
 				if _, ok := pvc.GetAnnotations()[rsCronJobScheduleTimeAnnotation]; ok {
 					continue
 				}
+				if _, ok := pvc.GetAnnotations()[krcJobScheduleTimeAnnotation]; ok {
```

Comment:
What if 
- A storageclass already has rsCronJob annotation
- PVCs already have rsCronJob annotation
- krcJob annotation is added to that same storageclass

I think we need a more elaborate check here which also considers annotations on chnaged storageclass.